### PR TITLE
Use MSVC compiler flags only with MSVC

### PR DESCRIPTION
**Pull request by mimi89999**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,8 +13,10 @@ set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
 
-# TODO: re-enable /GL and /LTCG when we get a bigger build machine
-add_compile_options(/await /bigobj /GR- /permissive-)
+if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
```
Comment (dlech) [inline CMakeLists.txt:16]: I think we can simplify this to the following:

```suggestion
if (MSVC)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -13,8 +13,10 @@ set(CMAKE_CXX_STANDARD 17)
...
+    # TODO: re-enable /GL and /LTCG when we get a bigger build machine
+    add_compile_options(/await /bigobj /GR- /permissive-)
+endif()
```
Comment (dlech) [inline CMakeLists.txt:19]: It wouldn't hurt to add an `else()` with `add_compile_options(-Werror)` for clang/gcc compile options.

File: CMakeLists.txt [hunk 3]
```diff
@@ -34,12 +36,14 @@ if($ENV{CI})
     set_property(TARGET _winrt PROPERTY JOB_POOL_COMPILE compile_job)
 endif()
 
-string(REGEX REPLACE "/GR" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
-# TODO: re-enable /GL and /LTCG when we get a bigger build machine
-# string(APPEND CMAKE_MODULE_LINKER_FLAGS " /LTCG:STATUS")
+if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
```
Comment (dlech) [inline CMakeLists.txt:39]: While we are touching this, let's move this into the same if statement above.

**Review by dlech — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 4]
```diff
@@ -13,8 +13,20 @@ set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
 
-# TODO: re-enable /GL and /LTCG when we get a bigger build machine
-add_compile_options(/await /bigobj /GR- /permissive-)
+if ("MSVC")
```
Comment (dlech) [inline CMakeLists.txt:16]: If it is in quotes, doesn't that make it a string (which is always false)? Shouldn't it be unquoted to use the variable name instead?

```suggestion
if (MSVC)
```

**Comment by dlech** [comment]
Nice, thanks.

**Comment by mimi89999** [comment]
@dlech Thanks for merging this. Could you please update the `pywinrt` bindings in this repo?

**Pull request merged**